chat_shell/chat_shell/tools/builtin/load_skill.py

```
@@ -120,8 +130,95 @@ async def _arun(
+        display_name = skill_info.get("displayName")
+        if display_name:
+            self._skill_display_names[skill_name] = display_name
+
+        # NEW: Trigger lazy loading of skill tools and MCP servers
+        await self._lazy_load_skill_resources(skill_name, skill_info)
+
+        # Return confirmation message
+        return f"Skill '{skill_name}' has been loaded. The instructions have been added to the system prompt. Please follow them strictly."
+
+    async def _lazy_load_skill_resources(
+        self, skill_name: str, skill_info: dict
```

Ruff (0.14.11)

Unused method argument: `skill_info`

(ARG002)